### PR TITLE
ci: tcp logger with real logger server

### PR DESCRIPTION
**Pull request by Sn0rt**
### Description

#9391

the [vector](https://vector.dev/docs/reference/configuration/sources/) is a popular logger service implemented  by  `Rust`

to add the vector as the logger server for receiving the log from the user client.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by bzp2010 — COMMENTED** [review]
For the content, I see no problem. But I have a question: is there any reason why you chose `vector`?

**Comment by Sn0rt** [comment]
> For the content, I see no problem. But I have a question: is there any reason why you chose `vector`?

Simple, easy to configure, well documented.

There is another reason why I tried to use Fluentd and encountered a strange case. I spent half a day trying to fix it to no avail. There is no problem with how to use `vector `.

**Comment by bzp2010** [comment]
> Simple, easy to configure, well documented.
> 
> There is another reason why I tried to use Fluentd and encountered a strange case. I spent half a day trying to fix it to no avail.

OK, @Sn0rt 

For the first point, that's very good. And for the second point, have you tried fluent bit?

**Comment by Sn0rt** [comment]
> > Simple, easy to configure, well documented.
> > There is another reason why I tried to use Fluentd and encountered a strange case. I spent half a day trying to fix it to no avail.
> 
> OK, @Sn0rt
> 
> For the first point, that's very good. And for the second point, have you tried fluent bit (it is the next generation of fluentd)?

I took a brief look at the document construction of fluent-bit, which is not as good as that of vector. My requirement is to cooperate with multiple reporting log formations. fluent-bit does not have an example available out of the box, but vector provides it.

In this Pr, I expect vector to become the basic component of a ci log plug-in test, and to cooperate with other log plug-ins later.

**Comment by bzp2010** [comment]
@Sn0rt Understand and recognize your position.

But I'm not sure if it's the right strategy to keep using `vector` in the future because essentially the open source project is a vendor-neutral project, the plugin is called `tcp-logger` and customers may use any logging pipeline project that supports TCP, I'm not sure....

**Review by bzp2010 — DISMISSED** [review]

**Review by shreemaan-abhishek — COMMENTED** [review]
LGTM

**Review by Neilblaze — APPROVED** [review]

File: ci/pod/vector/vector_logs_ca.key [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (soulbird) [inline ci/pod/vector/vector_logs_ca.key:1]: We'd better move certs to `t/certs`

Comment (Sn0rt) [reply ci/pod/vector/vector_logs_ca.key]: updated

File: t/plugin/tcp-logger.t [hunk 2]
```diff
@@ -482,3 +482,119 @@ hello world
...
+    }
+--- request
+GET /t
+--- wait: 0.5
+--- response_body
+opentracing
+
+
+
+=== TEST 15: check tcp log
```
Comment (monkeyDluffy6017) [inline t/plugin/tcp-logger.t:588]: The test case 15 should be put before the test case 14

Comment (Sn0rt) [reply t/plugin/tcp-logger.t]: good.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Comment by soulbird** [comment]
We need to adjust the existing test cases. Currently, the log is forwarded to port 5044, which is a simulated log service.

**Comment by Sn0rt** [comment]
> We need to adjust the existing test cases. Currently, the log is forwarded to port 5044, which is a simulated log service.

copy that. I rewrite some of the tcp-logger test cases.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**